proto/workflow_execution_service.proto

```diff
@@ -0,0 +1,237 @@
+// and executing a given workflow.
+message workflow_request {
+  // OPTIONAL
+  // The workflow CWL or WDL document, must provide either this or workflow_url
+  string workflow_descriptor = 1;
+
+  // REQUIRED
+  // The workflow parameterization document (typically a JSON string), includes all parameterizations for
+  // the workflow including input and output file locations
+  string workflow_params = 2;
+
+  // REQUIRED
+  // The workflow descriptor type, must be "CWL" or "WDL" currently (or another alternative supported by this WES instance)
+  string workflow_type = 3;
+
+  // REQUIRED
+  // The workflow descriptor type version, must be one supported by this WES instance
+  string workflow_type_version = 4;
+
+  // OPTIONAL
+  // A key-value map of arbitrary metadata outside the scope of the workflow_params but useful to track with this workflow request
+  map<string,string> key_values = 5;
+
+  // OPTIONAL
+  // The workflow CWL or WDL document, must provide either this or workflow_descriptor
+  string workflow_url = 6;
+}
+
+// Request listing of jobs tracked by server
+message workflow_list_request {
+  // OPTIONAL
+  // Number of workflows to return in a page.
+  uint32 page_size = 1;
+
+  // OPTIONAL
+  // Token to use to indicate where to start getting results. If unspecified, returns the first
+  // page of results.
+  string page_token = 2;
+
+  // OPTIONAL
+  // For each key, if the key's value is empty string then match workflows that are tagged with
+  // this key regardless of value
+  string key_value_search = 3;
+}
+
+
+// The service will return a workflow_list_response when receiving a successful workflow_list_request.
+message workflow_list_response {
+  // A list of workflows that the service has executed or is executing.
+  repeated workflow_desc workflows = 1;
+
+  // A token, which when provided in a workflow_list_request, allows one to retrieve the next page
+  // of results.
+  string next_page_token = 2;
+}
+
+// The service info request is a simple GET request and doesn't take any parameters.
+message service_info_request {}
+
+// A message containing useful information about the running service, including supported versions and
+// default settings.
+message service_info {
+  // A map with keys as the workflow format type name (currently only CWL and WDL are used
+  // although a service may support others) and value is a workflow_type_version object which
+  // simply contains an array of one or more version strings
+  map<string, workflow_type_version> workflow_type_versions = 1;
+
+  // The version(s) of the WES schema supported by this service
+  repeated string supported_wes_versions = 2;
+
+  // The filesystem protocols supported by this service, currently these may include common
+  // protocols such as 'http', 'https', 'sftp', 's3', 'gs', 'file', 'synapse', or others as
+  // supported by this service.
+  repeated string supported_filesystem_protocols = 3;
```

Comment:
We should have a discussion about syncing this concept with the one used in TES: https://github.com/ga4gh/task-execution-schemas/blob/master/task_execution.proto#L527

Reply:
I created an issue https://github.com/ga4gh/workflow-execution-schemas/issues/15